### ISSUE TITLE
F# Support?

### ISSUE THREAD

**Issue by Rickasaurus** [georgeduckett/TeamCoding#11]
This project looks really rad, but we're mostly an F# shop and so can't use it. Any chance for F# support with the new Roslyn stuff? https://github.com/dotnet/project-system/pull/1670

**Comment by georgeduckett** [comment]
TL;DR: You now get a caret position in the same way as a generic text file is handled, but not in the way VB.NET and C# are supported. I've updated the extension on the marketplace.

----
Depends what you mean by F# support. I'm assuming you were looking to get a caret position (when I tested it looked like it was tracking what documents people had open, but wasn't showing where they were in F# docs.

The caret position is calculated in two different ways, either by treating it as a simple text file and translating the local line number into a server line number (using local diffs) then back again, or it uses Roslyn to look at the syntax tree to find the member the user is in (e.g. the class' method) and uses that plus an offset.

What was happening is it was getting the Document from the project system for an F# doc, but it wasn't getting the syntax tree for an F# doc (and still don't think it does given your link above as that looks like F# project support, not F# syntax). I've changed the code to fall back to using the text/diff based method if it can't get the syntax tree (even if it can get the Document object). I've updated the marketplace with the latest version including this fix.
